core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -1634,6 +1655,86 @@ class TestMetricsReporter extends MetricsReporter with Reconfigurable with Close
   }
 }
 
+object TestNumReplicaFetcherMetricsReporter {
+  val testReporters = new ConcurrentLinkedQueue[TestNumReplicaFetcherMetricsReporter]()
+  val configuredBrokers = mutable.Set[Int]()
+
+  def waitForReporters(count: Int): List[TestNumReplicaFetcherMetricsReporter] = {
+    TestUtils.waitUntilTrue(() => testReporters.size == count, msg = "Metrics reporters size not matched. Expected: " + count + ", actual: " + testReporters.size())
+
+    val reporters = testReporters.asScala.toList
+    TestUtils.waitUntilTrue(() => reporters.forall(_.configureCount == 1), msg = "Metrics reporters not configured")
+    reporters
+  }
+}
+
+
+class TestNumReplicaFetcherMetricsReporter extends MetricsReporter with Reconfigurable with Closeable with ClusterResourceListener {
```

Comment:
The reporter's functionality is excessive. All we need to verify is that is can "see" the latest configurations at startup. Therefore, only `reconfigureCount` is relevant for the new test case